Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -58,17 +83,48 @@ namespace MultiplayerSample
         //Signal event to end the match
         m_roundTickEvent.RemoveFromQueue();
 
-        // TODO: continuously populate player state base on coins, etc.
-        PlayerState jackState = PlayerState{ "Jack", 25, 80 };
-        PlayerState allieState = PlayerState{ "Allie", 23, 70 };
-        PlayerState olexState = PlayerState{ "Olex", 5, 42 };
+        MatchResultsSummary results;
 
-        SetPlayerStates(0, jackState);
-        SetPlayerStates(1, allieState);
-        SetPlayerStates(2, olexState);
+        const AZStd::vector<PlayerCoinState>& coinStates = GetMatchPlayerCoinsComponentController()->GetParent().
+            GetPlayerCoinCounts();
 
-        // TODO: formulate and send real results
-        RPC_EndMatch(MatchResultsSummary{ "Jack", {jackState, allieState, olexState} });
+        int highestCoins = -1;
+
+        for (const Multiplayer::NetEntityId playerNetEntity : m_players)
+        {
+            PlayerState state;
+            const auto playerHandle = Multiplayer::GetNetworkEntityManager()->GetEntity(playerNetEntity);
+            if (playerHandle.Exists())
+            {
+                if (const PlayerIdentityComponent* identity = playerHandle.GetEntity()->FindComponent<PlayerIdentityComponent>())
+                {
+                    state.m_playerName = identity->GetPlayerName();
+                }
+                if (const NetworkHealthComponent* armor = playerHandle.GetEntity()->FindComponent<NetworkHealthComponent>())
+                {
+                    // Treating health as armor
+                    state.m_remainingShield = aznumeric_cast<uint8_t>(armor->GetHealth());
```

Comment:
should we rename `m_remainingShield` to `m_remainingArmor` ? Already on this one line we're calling it 3 different things (shield, armor, health), I think it would help to eliminate one variation at least.

Reply:
I wasn't sure where shield came from, so I kept it as is. But we should definitely bring it up on the next sync and check on the terms from the design docs.